FizzBuzz 1/FizzBuzz/Program.cs

```diff
@@ -1,42 +1,117 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace FizzBuzz
 {
     class Program
     {
+        static Boolean ReadYN(string message)
+        {
+            char answer = 'a';
+            var valid = new List<char>();
+            valid.Add('y');
+            valid.Add('n');
+            while (!valid.Contains(answer))
+            {
+                Console.Write(message);
+                try
+                {
+                    answer = Char.Parse(Console.ReadLine());
+                    answer = Char.ToLower(answer);
+                }
+                catch
+                {
+                }
+            }
+            if (answer == 'y')
+                return true;
+            else
+                return false;
+        }
+
         static void Main(string[] args)
         {
-            for (int i = 1; i <= 260; i++)
+            int max = -1;
+            while (max < 1)
+            {
+                Console.Write("Enter a number to go up to: ");
+                try
+                {
+                    max = Int32.Parse(Console.ReadLine());
+                } catch
+                {
+                }
+            }
+
+            Boolean[] rules = new bool[6];
+
+            if (ReadYN("Enable Fizz if mod 3? (y/n): "))
```

Comment:
Can you simplify these blocks so that they're just an assignment instead of an assignment inside an `if`? 🙂